### PR TITLE
Update driver: Exception, EditDriverCommand, Driver Status

### PR DESCRIPTION
**Pull request by hieunm1821**

### CODE REVIEW

**Review by jaegarpoon — APPROVED** [review]

**Comment by jaegarpoon** [comment]
Looks good! Like the use of enums :)

File: src/main/java/seedu/address/logic/commands/EditDriverCommand.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        if (!driverToEdit.isSameDriver(editedDriver) && model.hasDriver(editedDriver)) {
+            throw new CommandException(MESSAGE_DUPLICATE_DRIVER);
+        }
+
+        if (driverToEdit.isBusy()) {
+            throw new CommandException(String.format(MESSAGE_DELETE_DRIVER_FAIL_BUSY, driverToEdit));
+        }
+
+        model.setDriver(driverToEdit, editedDriver);
+        model.updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
```
Comment (xzzz3) [inline src/main/java/seedu/address/logic/commands/EditDriverCommand.java:82]: this line should be the driver class version of it

File: src/main/java/seedu/address/logic/commands/EditDriverCommand.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        // state check
+        EditDriverCommand e = (EditDriverCommand) other;
+        return index.equals(e.index)
+                && editDriverDescriptor.equals(e.editDriverDescriptor);
+    }
+
+
+    /**
+     * Stores the details to edit the driver with each non-empty field value will replace the
+     * corresponding field value of the customer.
```
Comment (xzzz3) [inline src/main/java/seedu/address/logic/commands/EditDriverCommand.java:124]: customer --> driver

**Review by xzzz3 — APPROVED** [review]
LGTM!

**Pull request merged**